src/main/java/Duke.java

```diff
@@ -1,10 +1,204 @@
+                //description[1] to get description - duke
+                String[] taskDescription = str[0].split(" ", 2);
+                SimpleDateFormat dateFormat = new SimpleDateFormat("dd MMM yyyy");
+                //Format date 12 Oct 2019, to string "12 Oct 2019"
+                Date date = dateFormat.parse(str[1]);
+                String strDate = dateFormat.format(date);
+                Task t = new Deadline(taskDescription[1], strDate);
+                taskList.add(t);
+            } else {
+                //e.g deadline duke
+                String taskDescription = line.substring(9);
+                Task t = new Deadline(taskDescription, "Date not specified");
+                taskList.add(t);
+            }
+            UI.printTask(taskList);
+        } catch (EmptyDeadlineDescriptionException e) {
+            UI.printEmptyDeadlineDescriptionException();
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+    }
+
+    static void list() {
+        try {
+            listEmpty(taskList);
+            UI.printOutput((taskList));
+        } catch (ListEmptyException e) {
+            UI.printListEmpty();
+        }
+    }
+
+    static void done(String line) {
+        try {
+            listEmpty(taskList);
+            String theStr = line.substring(5);
+            String[] strArr = theStr.split(",");
+            int[] intArr = new int[strArr.length];
+            for (int i = 0; i < strArr.length; i++) {
+                String num = strArr[i];
+                intArr[i] = Integer.parseInt(num);
+                indexOutOfRange(taskList.size(), intArr[i]);
+            }
+
+            for (int i = 0; i < intArr.length; i++) {
+                Task t = taskList.get(intArr[i] - 1);
+                t.markAsDone();
+                UI.printMarkedAsDone(t);
+            }
+        } catch (NumberFormatException e) {
+            UI.printNumberFormatException();
+        }
+        catch (IndexOutOfRangeException e) {
+            UI.printIndexOutOfRangeException();
+        }
+        catch  (ListEmptyException e) {
+            UI.printListEmpty();
+        }
+    }
+
+    static void invalidTask(String line) {
+        try {
+            checkEmpty(line);
+            checkInvalidWord(line);
+        } catch (EmptyException e) {
+            UI.printEmptyException();
+        } catch (StringFormatException e){
+            UI.printStringFormatException();
+        }
+    }
+
+    static void checkEmpty(String description) throws EmptyException {
+        if (description.isEmpty()){
+            throw new EmptyException ();
+        }
+    }
+
+    static void checkInvalidWord(String description) throws StringFormatException {
+        if ( !( description.equals("bye") || description.equals("list") || description.equals("todo") || description.equals("event") || description.equals("deadline"))) {
+            throw new StringFormatException ();
```

Comment:
Think here one can simplify the logic statements further as suggested in Maximise readability guidelines?

Reply:
Thanks for the advice, will improvise the code accordingly!